rotkehlchen/chain/bitcoin/utils.py

```diff
@@ -107,21 +188,91 @@ def pubkey_to_p2sh_p2wpkh_address(data: bytes) -> BTCAddress:
 
 def pubkey_to_bech32_address(data: bytes, witver: int) -> BTCAddress:
     """
-    Bitcoin pubkey to bech32 address
+    Bitcoin pubkey to native Segwit(P2WPKH) & Taproot(P2TR) addresses.
 
     Source:
     https://github.com/bitcoin/bips/blob/master/bip-0141.mediawiki#witness-program
-    https://github.com/mcdallas/cryptotools/blob/master/btctools/address.py
+    https://github.com/bitcoin/bips/blob/master/bip-0086.mediawiki
 
     May raise:
     - EncodingError if address could not be derived from public key
     """
-    witprog = hash160(data)
-    result = bech32.encode('bc', witver, witprog)
```

Comment:
Hm... looks a bit suspicious that previously this function was already supporting different `witver` values. So what is the difference between new logic and passing `witver=1` to this function in the past?

Reply:
Throughout the codebase iirc, only `witver=0` was passed.

The major difference is the `XOR 1` & `XOR 0x2bc830a3` which represents version 0 & > 1 respectively used in `_bech32_create_checksum`

If you see `bech32.bech32_create_checksum`, only `XOR 1`  or `^ 1` is seen.

Reply:
Ah, got it, thanks